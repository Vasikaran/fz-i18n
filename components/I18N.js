import React,{PropTypes, Children} from 'react';

function replaceI18NValuesWithRegex(i18nStr,values)
{
	if(typeof values !== "undefined")
	{
		if(Array.isArray(values)){
			for(var i=0;i< values.length;i++){
				i18nStr = i18nStr.replace( new RegExp( "\\{"+i+"\\}","g") ,values[i]);
			}
		}
		else {
			i18nStr = i18nStr.replace( new RegExp( "\\{0\\}","g") ,values);
		}
	}
	return i18nStr;
}
function unescapeUnicode(str)
{
    return str.replace( /\\u([a-fA-F0-9]{4})/g, function(g, m1) {
         return String.fromCharCode(parseInt(m1, 16));
    });
}	

export default class I18N extends React.Component {  
	constructor(props) {
		super(props);
		this.getI18NValue=this.getI18NValue.bind(this);
		this.createElement=this.createElement.bind(this);
	}
	getI18NValue(){
		const {i18NKey:key,values} =this.props;
		const {i18n} =this.context;
		console.log(values)
		console.log("getI18NValue",key);
		if(typeof i18n === "undefined" || key === "" || typeof key === "undefined"){
			return key;
		}
		var i18nStr = i18n[key];
		if(i18nStr === undefined){
			return key;
		}
		i18nStr = replaceI18NValuesWithRegex(i18nStr,values);
		return unescapeUnicode(i18nStr);
	}
	createElement(){
		const props = Object.keys(this.props).reduce((result,nextKey)=>{
			if(nextKey!="i18NKey" && nextKey!="tag" && nextKey!="values")
			result[nextKey]=this.props[nextKey];
			return result;
		},{});
		const child=this.getI18NValue();
		return React.createElement(this.props.tag,
			props,
			this.getI18NValue()
		);
	}
  	
  render() {
    return this.createElement();
  }
}
I18N.propTypes={
	i18NKey:PropTypes.object.isRequired,
	values:PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.array
	]),
	tag:PropTypes.string
}
I18N.defaultProps={
	tag:"span"
}
I18N.contextTypes = {
  i18n: PropTypes.object
}

