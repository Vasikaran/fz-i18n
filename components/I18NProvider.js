import React,{PropTypes, Children} from 'react';

const emptyObj={};
export class I18NProvider extends React.Component{
	getChildContext() {
    return {i18n: this.props.i18n};
  }
  render(){
  	return Children.only(this.props.children)
  }

}

I18NProvider.defaultProps = {
	i18n:emptyObj
}
I18NProvider.childContextTypes = {
  i18n: PropTypes.object
};

export default I18N;