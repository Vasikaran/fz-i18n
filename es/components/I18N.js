import _Object$assign from 'babel-runtime/core-js/object/assign';
import _Object$keys from 'babel-runtime/core-js/object/keys';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { replaceI18NValuesWithRegex, unescapeUnicode } from '../utils';

var I18N = function (_React$Component) {
  _inherits(I18N, _React$Component);

  function I18N(props) {
    _classCallCheck(this, I18N);

    var _this = _possibleConstructorReturn(this, (I18N.__proto__ || _Object$getPrototypeOf(I18N)).call(this, props));

    _this.getI18NValue = _this.getI18NValue.bind(_this);
    _this.createElement = _this.createElement.bind(_this);
    return _this;
  }

  _createClass(I18N, [{
    key: 'getI18NValue',
    value: function getI18NValue() {
      var _props = this.props,
          key = _props.i18NKey,
          values = _props.values;
      var i18n = this.context.i18n;

      if (typeof i18n === 'undefined') {
        return key;
      }
      var i18nStr = i18n[key];
      if (i18nStr === undefined) {
        return key;
      }
      i18nStr = replaceI18NValuesWithRegex(i18nStr, values);
      return unescapeUnicode(i18nStr);
    }
  }, {
    key: 'createElement',
    value: function createElement() {
      var _this2 = this;

      var props = _Object$keys(this.props).reduce(function (result, nextKey) {
        if (nextKey != 'i18NKey' && nextKey != 'tag' && nextKey != 'values' && nextKey != 'isHtml') result[nextKey] = _this2.props[nextKey];
        return result;
      }, {});
      //const child=this.getI18NValue();
      if (this.props.isHtml) {
        var dangerouslySetInnerHTML = {
          __html: this.getI18NValue()
        };

        return React.createElement(this.props.tag, _Object$assign(props, { dangerouslySetInnerHTML: dangerouslySetInnerHTML }));
      } else {
        return React.createElement(this.props.tag, props, this.getI18NValue());
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.createElement();
    }
  }]);

  return I18N;
}(React.Component);

export default I18N;

I18N.propTypes = {
  i18NKey: PropTypes.string.isRequired,
  values: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  tag: PropTypes.string,
  isHtml: PropTypes.bool
};
I18N.defaultProps = {
  tag: 'span',
  isHtml: false
};
I18N.contextTypes = {
  i18n: PropTypes.object
};