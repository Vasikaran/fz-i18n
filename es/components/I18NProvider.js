import _typeof from 'babel-runtime/helpers/typeof';
import _Promise from 'babel-runtime/core-js/promise';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getI18NValue, userDateFormat } from '../utils';

var emptyObj = {};
var dummy = function dummy(key, values) {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
export var i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};

var I18NProvider = function (_React$Component) {
  _inherits(I18NProvider, _React$Component);

  function I18NProvider(props, context) {
    _classCallCheck(this, I18NProvider);

    var _this = _possibleConstructorReturn(this, (I18NProvider.__proto__ || _Object$getPrototypeOf(I18NProvider)).call(this, props, context));

    _this.i18n = props.i18n;
    _this.timeZone = props.timeZone;
    _this.user = props.user;
    _this.permission = props.permission;
    _this.license = props.license;
    _this.promise = null;
    _this.resolve = null;
    _this.reject = null;
    _this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = getI18NValue(props.i18n);
    i18NProviderUtils.userDateFormat = userDateFormat(i18NProviderUtils.getI18NValue, props.timeZone);
    return _this;
  }

  _createClass(I18NProvider, [{
    key: 'getChildContext',
    value: function getChildContext() {
      return {
        i18n: this.i18n,
        timeZone: this.timeZone,
        user: this.user,
        permission: this.permission,
        license: this.license
      };
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var _this2 = this;

      if (this.props.i18n != nextProps.i18n || this.props.timeZone != nextProps.timeZone || this.props.user != nextProps.user || this.props.permission != nextProps.permission || this.props.license != nextProps.license && this.promise) {
        this.promise = new _Promise(function (res, rej) {
          _this2.resolve = res;
          _this2.reject = rej;
        }).then(function () {
          _this2.i18n = nextProps.i18n;
          _this2.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = getI18NValue(nextProps.i18n);
          i18NProviderUtils.userDateFormat = userDateFormat(i18NProviderUtils.getI18NValue, nextProps.timeZone);
          _this2.setState({ reRender: true }, function () {
            setTimeout(function () {
              return _this2.setState({ reRender: false });
            }, 1);
          });
          _this2.promise = null;
        }, function () {
          _this2.promise = null;
        });
        this.props.onChange(this.resolve, this.reject);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.state.reRender ? null : Children.only(this.props.children);
    }
  }]);

  return I18NProvider;
}(React.Component);

export default I18NProvider;


I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: PropTypes.object,
  timeZone: PropTypes.string,
  user: PropTypes.object,
  permission: PropTypes.object,
  license: PropTypes.object
};

export var ConnectI18NProvider = connect(function (state, props) {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n: i18n,
    timeZone: timeZone
  };
})(I18NProvider);

export var AppProvider = connect(function (state, props) {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (_typeof(props.user) == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (_typeof(props.permission) == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (_typeof(props.license) == 'object') {
    license = props.license;
  }
  return {
    i18n: i18n,
    timeZone: timeZone,
    user: user,
    permission: permission,
    license: license
  };
})(I18NProvider);