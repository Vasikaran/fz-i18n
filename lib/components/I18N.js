'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var I18N = function (_React$Component) {
  _inherits(I18N, _React$Component);

  function I18N(props) {
    _classCallCheck(this, I18N);

    var _this = _possibleConstructorReturn(this, (I18N.__proto__ || Object.getPrototypeOf(I18N)).call(this, props));

    _this.getI18NValue = _this.getI18NValue.bind(_this);
    _this.createElement = _this.createElement.bind(_this);
    return _this;
  }

  _createClass(I18N, [{
    key: 'getI18NValue',
    value: function getI18NValue() {
      var _props = this.props,
          key = _props.i18NKey,
          values = _props.values;
      var i18n = this.context.i18n;

      if (typeof i18n === 'undefined') {
        return key;
      }
      var i18nStr = i18n[key];
      if (i18nStr === undefined) {
        return key;
      }
      i18nStr = (0, _utils.replaceI18NValuesWithRegex)(i18nStr, values);
      return (0, _utils.unescapeUnicode)(i18nStr);
    }
  }, {
    key: 'createElement',
    value: function createElement() {
      var _this2 = this;

      var props = Object.keys(this.props).reduce(function (result, nextKey) {
        if (nextKey != 'i18NKey' && nextKey != 'tag' && nextKey != 'values' && nextKey != 'isHtml') result[nextKey] = _this2.props[nextKey];
        return result;
      }, {});
      //const child=this.getI18NValue();
      if (this.props.isHtml) {
        var dangerouslySetInnerHTML = {
          __html: this.getI18NValue()
        };

        return _react2.default.createElement(this.props.tag, Object.assign(props, { dangerouslySetInnerHTML: dangerouslySetInnerHTML }));
      } else {
        return _react2.default.createElement(this.props.tag, props, this.getI18NValue());
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.createElement();
    }
  }]);

  return I18N;
}(_react2.default.Component);

exports.default = I18N;

I18N.propTypes = {
  i18NKey: _propTypes2.default.string.isRequired,
  values: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.array]),
  tag: _propTypes2.default.string,
  isHtml: _propTypes2.default.bool
};
I18N.defaultProps = {
  tag: 'span',
  isHtml: false
};
I18N.contextTypes = {
  i18n: _propTypes2.default.object
};