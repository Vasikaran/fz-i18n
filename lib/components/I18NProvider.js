'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AppProvider = exports.ConnectI18NProvider = exports.i18NProviderUtils = undefined;

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var emptyObj = {};
var dummy = function dummy(key, values) {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
var i18NProviderUtils = exports.i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};

var I18NProvider = function (_React$Component) {
  _inherits(I18NProvider, _React$Component);

  function I18NProvider(props, context) {
    _classCallCheck(this, I18NProvider);

    var _this = _possibleConstructorReturn(this, (I18NProvider.__proto__ || Object.getPrototypeOf(I18NProvider)).call(this, props, context));

    _this.i18n = props.i18n;
    _this.timeZone = props.timeZone;
    _this.user = props.user;
    _this.permission = props.permission;
    _this.license = props.license;
    _this.promise = null;
    _this.resolve = null;
    _this.reject = null;
    _this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = (0, _utils.getI18NValue)(props.i18n);
    i18NProviderUtils.userDateFormat = (0, _utils.userDateFormat)(i18NProviderUtils.getI18NValue, props.timeZone);
    return _this;
  }

  _createClass(I18NProvider, [{
    key: 'getChildContext',
    value: function getChildContext() {
      return {
        i18n: this.i18n,
        timeZone: this.timeZone,
        user: this.user,
        permission: this.permission,
        license: this.license
      };
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var _this2 = this;

      if (this.props.i18n != nextProps.i18n || this.props.timeZone != nextProps.timeZone || this.props.user != nextProps.user || this.props.permission != nextProps.permission || this.props.license != nextProps.license && this.promise) {
        this.promise = new Promise(function (res, rej) {
          _this2.resolve = res;
          _this2.reject = rej;
        }).then(function () {
          _this2.i18n = nextProps.i18n;
          _this2.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = (0, _utils.getI18NValue)(nextProps.i18n);
          i18NProviderUtils.userDateFormat = (0, _utils.userDateFormat)(i18NProviderUtils.getI18NValue, nextProps.timeZone);
          _this2.setState({ reRender: true }, function () {
            setTimeout(function () {
              return _this2.setState({ reRender: false });
            }, 1);
          });
          _this2.promise = null;
        }, function () {
          _this2.promise = null;
        });
        this.props.onChange(this.resolve, this.reject);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.state.reRender ? null : _react.Children.only(this.props.children);
    }
  }]);

  return I18NProvider;
}(_react2.default.Component);

exports.default = I18NProvider;


I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: _propTypes2.default.object,
  timeZone: _propTypes2.default.string,
  user: _propTypes2.default.object,
  permission: _propTypes2.default.object,
  license: _propTypes2.default.object
};

var ConnectI18NProvider = exports.ConnectI18NProvider = (0, _reactRedux.connect)(function (state, props) {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n: i18n,
    timeZone: timeZone
  };
})(I18NProvider);

var AppProvider = exports.AppProvider = (0, _reactRedux.connect)(function (state, props) {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (_typeof(props.user) == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (_typeof(props.permission) == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (_typeof(props.license) == 'object') {
    license = props.license;
  }
  return {
    i18n: i18n,
    timeZone: timeZone,
    user: user,
    permission: permission,
    license: license
  };
})(I18NProvider);