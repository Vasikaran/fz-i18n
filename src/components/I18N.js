import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { replaceI18NValuesWithRegex, unescapeUnicode } from '../utils';

export default class I18N extends React.Component {
  constructor(props) {
    super(props);
    this.getI18NValue = this.getI18NValue.bind(this);
    this.createElement = this.createElement.bind(this);
  }
  getI18NValue() {
    const { i18NKey: key, values } = this.props;
    const { i18n } = this.context;
    if (typeof i18n === 'undefined') {
      return key;
    }
    var i18nStr = i18n[key];
    if (i18nStr === undefined) {
      return key;
    }
    i18nStr = replaceI18NValuesWithRegex(i18nStr, values);
    return unescapeUnicode(i18nStr);
  }
  createElement() {
    const props = Object.keys(this.props).reduce((result, nextKey) => {
      if (
        nextKey != 'i18NKey' &&
        nextKey != 'tag' &&
        nextKey != 'values' &&
        nextKey != 'isHtml'
      )
        result[nextKey] = this.props[nextKey];
      return result;
    }, {});
    //const child=this.getI18NValue();
    if (this.props.isHtml) {
      var dangerouslySetInnerHTML = {
        __html: this.getI18NValue()
      };

      return React.createElement(
        this.props.tag,
        Object.assign(props, { dangerouslySetInnerHTML })
      );
    } else {
      return React.createElement(this.props.tag, props, this.getI18NValue());
    }
  }

  render() {
    return this.createElement();
  }
}
I18N.propTypes = {
  i18NKey: PropTypes.string.isRequired,
  values: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  tag: PropTypes.string,
  isHtml: PropTypes.bool
};
I18N.defaultProps = {
  tag: 'span',
  isHtml: false
};
I18N.contextTypes = {
  i18n: PropTypes.object
};
