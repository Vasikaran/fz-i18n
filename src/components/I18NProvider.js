import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getI18NValue, userDateFormat } from '../utils';

const emptyObj = {};
const dummy = (key, values) => {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
export const i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};
export default class I18NProvider extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.i18n = props.i18n;
    this.timeZone = props.timeZone;
    this.user = props.user;
    this.permission = props.permission;
    this.license = props.license;
    this.promise = null;
    this.resolve = null;
    this.reject = null;
    this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = getI18NValue(props.i18n);
    i18NProviderUtils.userDateFormat = userDateFormat(
      i18NProviderUtils.getI18NValue,
      props.timeZone
    );
  }
  getChildContext() {
    return {
      i18n: this.i18n,
      timeZone: this.timeZone,
      user: this.user,
      permission: this.permission,
      license: this.license
    };
  }
  componentWillReceiveProps(nextProps) {
    if (
      this.props.i18n != nextProps.i18n ||
      this.props.timeZone != nextProps.timeZone ||
      this.props.user != nextProps.user ||
      this.props.permission != nextProps.permission ||
      (this.props.license != nextProps.license && this.promise)
    ) {
      this.promise = new Promise((res, rej) => {
        this.resolve = res;
        this.reject = rej;
      }).then(
        () => {
          this.i18n = nextProps.i18n;
          this.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = getI18NValue(nextProps.i18n);
          i18NProviderUtils.userDateFormat = userDateFormat(
            i18NProviderUtils.getI18NValue,
            nextProps.timeZone
          );
          this.setState({ reRender: true }, () => {
            setTimeout(() => this.setState({ reRender: false }), 1);
          });
          this.promise = null;
        },
        () => {
          this.promise = null;
        }
      );
      this.props.onChange(this.resolve, this.reject);
    }
  }
  render() {
    return this.state.reRender ? null : Children.only(this.props.children);
  }
}

I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: PropTypes.object,
  timeZone: PropTypes.string,
  user: PropTypes.object,
  permission: PropTypes.object,
  license: PropTypes.object
};

export const ConnectI18NProvider = connect((state, props) => {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (typeof props.i18n == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (typeof props.timeZone == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n,
    timeZone
  };
})(I18NProvider);

export const AppProvider = connect((state, props) => {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (typeof props.i18n == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (typeof props.timeZone == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (typeof props.user == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (typeof props.permission == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (typeof props.license == 'object') {
    license = props.license;
  }
  return {
    i18n,
    timeZone,
    user,
    permission,
    license
  };
})(I18NProvider);
