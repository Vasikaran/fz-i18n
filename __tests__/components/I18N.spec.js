import I18N from '../../src/components/I18N';
import I18NProvider from '../../src/components/I18NProvider';
import React from 'react';
import renderer from 'react-test-renderer';

const defaultProps ={
	i18NKey:"key1"
} 
describe('I18N component', () => {
  it("Should display i18n value", ()=> { 
   var ele=renderer.create(
    <I18NProvider i18n={{key1:"vimal1"}}>
      <I18N {...defaultProps}/>
    </I18NProvider>
  );
    var tree=ele.toJSON();
    expect(tree).toMatchSnapshot();
  	
  })
 
})