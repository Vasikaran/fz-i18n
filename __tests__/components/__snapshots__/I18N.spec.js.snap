exports[`I18N component Should display i18n value 1`] = `
<span>
  vimal1
</span>
`;
